### PR TITLE
Write Logistic Regression documentation

### PR DESCRIPTION
**Pull request by avolkov-intel**
# Description
Add Logistic Regression and Optimizers chapters to oneDAL documentation

### CODE REVIEW

File: docs/source/onedal/algorithms/optimizers/newton-cg.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Computing
+---------
+
+Newton-CG optimizer also known as the hessian-free optimizer is used to minimize convex function 
+without calculating the hessian matrix but using a hessian product matrix operator. The descent direction
+in the Newton method is calculated using the formula :math:`d_k = -H_k^{-1} g_k` where :math:`g_k, H_k` are
+the gradient and hessian matrix of the loss function on the :math:`k`-th iteration. In Newton-CG method
+Conjugate Gradients solver is used to find the approximate solution to the equation :math:`H_k d_k = -g_k`.
+Conjugate Gradients solver can find solution to the system of linear equations :math:`Ax = b` taking
```
Comment (aepanchi) [inline docs/source/onedal/algorithms/optimizers/newton-cg.rst:41]:
```suggestion
This solver can find solutions to the system of linear equations :math:`Ax = b` taking 
```

File: docs/source/onedal/algorithms/optimizers/newton-cg.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+in the Newton method is calculated using the formula :math:`d_k = -H_k^{-1} g_k` where :math:`g_k, H_k` are
+the gradient and hessian matrix of the loss function on the :math:`k`-th iteration. In Newton-CG method
+Conjugate Gradients solver is used to find the approximate solution to the equation :math:`H_k d_k = -g_k`.
+Conjugate Gradients solver can find solution to the system of linear equations :math:`Ax = b` taking 
+vector :math:`b` and functor :math:`f(p) = Ap` as input.  
+
+
+.. _newton_cg_c_math_dense:
+
+Computation method: *dense*
```
Comment (aepanchi) [inline docs/source/onedal/algorithms/optimizers/newton-cg.rst:47]:
```suggestion
Computation Method: *dense*
```

File: docs/source/onedal/algorithms/optimizers/newton-cg.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+---------------------------
+The method defines the Newton-CG optimizer, which is used in other algorithms
```
Comment (aepanchi) [inline docs/source/onedal/algorithms/optimizers/newton-cg.rst:49]:
```suggestion
The method defines the Newton-CG optimizer used by other algorithms
```

File: docs/source/onedal/algorithms/optimizers/newton-cg.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+for convex optimization. There are no separate computation modes to minimize
```
Comment (aepanchi) [inline docs/source/onedal/algorithms/optimizers/newton-cg.rst:50]:
```suggestion
for convex optimization. There is no separate computation mode to minimize
```

File: docs/source/onedal/algorithms/optimizers/newton-cg.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+function manually.
```
Comment (aepanchi) [inline docs/source/onedal/algorithms/optimizers/newton-cg.rst:51]:
```suggestion
a function manually.
```

**Review by aepanchi — APPROVED** [review]
LGTM

**Comment by avolkov-intel** [comment]
@Mergifyio copy rls/2024.0.1-rls

**Pull request merged**